Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -445,14 +450,33 @@ static MTLVertexStepFunction stepFunction(WGPUVertexStepMode stepMode)
     });
 }
 
-RenderPipeline::RenderPipeline(id<MTLRenderPipelineState> renderPipelineState, MTLPrimitiveType primitiveType, std::optional<MTLIndexType> indexType, MTLWinding frontFace, MTLCullMode cullMode, id<MTLDepthStencilState> depthStencilState, uint32_t vertexShaderInputBufferCount, Device& device)
+RenderPipeline::RenderPipeline(id<MTLRenderPipelineState> renderPipelineState, MTLPrimitiveType primitiveType, std::optional<MTLIndexType> indexType, MTLWinding frontFace, MTLCullMode cullMode, id<MTLDepthStencilState> depthStencilState, MTLRenderPipelineReflection *reflection, uint32_t vertexShaderInputBufferCount, Device& device)
+    : m_renderPipelineState(renderPipelineState)
+    , m_device(device)
+    , m_primitiveType(primitiveType)
+    , m_indexType(indexType)
```

Comment:
Wow this patch is doing a lot

Reply:
It does build on https://github.com/WebKit/WebKit/pull/8042

only top commit needs review after the first one is merged https://github.com/WebKit/WebKit/commit/17cd5f800c0db82254db1f0bed578f0e9c094616